tools/crm_error.c

```diff
@@ -56,12 +56,83 @@ get_strings(int rc, const char **name, const char **str)
     }
 }
 
+PCMK__OUTPUT_ARGS("result-code", "int", "char *", "char *");
+static int
+result_code_default(pcmk__output_t *out, va_list args)
+{
+    int code = va_arg(args, int);
+    char *name = va_arg(args, char *);
+    char *desc = va_arg(args, char *);
+
+    if (out->is_quiet(out)) {
+        /* If out->is_quiet(), we don't print the code -- so if name and desc
+         * are NULL, there's nothing to print. It makes no sense to call the
+         * formatter that way, but it's probably not worth an assert.
+        */
+        if ((name != NULL) && (desc != NULL)) {
+            pcmk__formatted_printf(out, "%s - %s\n", name, desc);
+        } else if (name != NULL) {
+            pcmk__formatted_printf(out, "%s\n", name);
+        } else if (desc != NULL) {
+            pcmk__formatted_printf(out, "%s\n", desc);
```

Comment:
Why not use out->info()?

Reply:
This was probably a remnant of a different approach. `out->info()` would be suitable.

Reply:
Fixed

Reply:
Now I remember. It's because we tell the formatter to use this compact format (instead of the columnar format that we use for lists) by setting `out->quiet = true`.

When `out->quiet` is true for a text output object, `out->info()` is a no-op. `pcmk__formatted_printf()` forces it to print the output. There's a good bit of precedent for that in `pcmk_output.c` already.

Reply:
That's risky since this is used as the default implementation, not just the text implementation. (What happens with --output-as=none?)

Not sure what the best alternative is though

Reply:
Good catch. That's a bug between the way `pcmk__formatted_printf()` is written and the way it's used... it's defined in `output_text.c` but it's used in several default formatter functions. So it can also get used with output type none.

Added a patch in #2847 
https://github.com/ClusterLabs/pacemaker/pull/2847/commits/b4768fa7db2df181328babcea89a61c66d19dfe5#r961161871

@clumens FYI